testcases/OpTestEnergyScale.py

```diff
@@ -33,41 +33,38 @@
 import re
 import sys
 
-from common.OpTestBMC import OpTestBMC
-from common.OpTestIPMI import OpTestIPMI
-from common.OpTestConstants import OpTestConstants as BMC_CONST
-from common.OpTestError import OpTestError
-from common.OpTestHost import OpTestHost
-from common.OpTestSystem import OpTestSystem
+import unittest
+
+import OpTestConfiguration
 from common.OpTestUtil import OpTestUtil
+from common.OpTestSystem import OpSystemState
+from common.OpTestConstants import OpTestConstants as BMC_CONST
 
+class OpTestEnergyScale(unittest.TestCase):
+    def setUp(self):
+        conf = OpTestConfiguration.conf
+        self.cv_HOST = conf.host()
+        self.cv_IPMI = conf.ipmi()
+        self.cv_SYSTEM = conf.system()
+        self.util = OpTestUtil()
+        self.cv_PLATFORM = conf.platform()
 
-class OpTestEnergyScale():
-    ##  Initialize this object
-    #  @param i_bmcIP The IP address of the BMC
-    #  @param i_bmcUser The userid to log into the BMC with
-    #  @param i_bmcPasswd The password of the userid to log into the BMC with
-    #  @param i_bmcUserIpmi The userid to issue the BMC IPMI commands with
-    #  @param i_bmcPasswdIpmi The password of BMC IPMI userid
-    #  @param i_ffdcDir Optional param to indicate where to write FFDC
+    ##
+    # @brief  It will execute and test the return code of ipmi command.
     #
-    # "Only required for inband tests" else Default = None
-    # @param i_hostIP The IP address of the HOST
-    # @param i_hostuser The userid to log into the HOST
-    # @param i_hostPasswd The password of the userid to log into the HOST with
+    # @param i_cmd @type string:The ipmitool command, for example: chassis power on; echo $?
     #
-    def __init__(self, i_bmcIP, i_bmcUser, i_bmcPasswd,
-                 i_bmcUserIpmi, i_bmcPasswdIpmi, i_ffdcDir=None, i_platName=None, i_hostip=None,
-                 i_hostuser=None, i_hostPasswd=None):
-        self.cv_BMC = OpTestBMC(i_bmcIP, i_bmcUser, i_bmcPasswd, i_ffdcDir)
-        self.cv_IPMI = OpTestIPMI(i_bmcIP, i_bmcUserIpmi, i_bmcPasswdIpmi,
-                                  i_ffdcDir, i_hostip, i_hostuser, i_hostPasswd)
-        self.cv_HOST = OpTestHost(i_hostip, i_hostuser, i_hostPasswd, i_bmcIP)
-        self.cv_SYSTEM = OpTestSystem(i_bmcIP, i_bmcUser, i_bmcPasswd,
-                         i_bmcUserIpmi, i_bmcPasswdIpmi, i_ffdcDir, i_hostip,
-                         i_hostuser, i_hostPasswd)
-        self.cv_PLATFORM = i_platName
-        self.util = OpTestUtil()
+    # @return l_res @type list: output of command or raise OpTestError
+    #
+    def run_ipmi_cmd(self, i_cmd):
+        l_cmd = i_cmd
+        l_res = self.cv_IPMI.ipmitool_execute_command(l_cmd)
+        print l_res
+        l_res = l_res.splitlines()
+        if int(l_res[-1]):
+            l_msg = "IPMI: command failed %c" % l_cmd
+            raise OpTestError(l_msg)
```

Comment:
fixed in 44178f7887505aec35c662129ac3416b684044fe, by importing OpTestError. Longer term, we should throw our own exception